dds/DCPS/DataSampleHeader.h

```diff
@@ -172,6 +172,9 @@ struct OpenDDS_Dcps_Export DataSampleHeader : public PoolAllocationBase {
   ACE_UINT32 lifespan_duration_nanosec_;  // Corresponding IDL is unsigned.
   //@}
 
+  /// Fragment size used by this sample
+  ACE_UINT32 fragment_size_;
```

Comment:
What's the equivalent of this in the code as it is today? Is there another place this could be stored so we don't have to change DataSampleHeader?

Reply:
I don't think there's an equivalent today, because the reader side doesn't have a need to know the fragment size. In general. It only worries about the fragment range associated with a buffer, and then tries to stitch the message back together based on those ranges. But now that we potentially need to split the buffer based on fragment, we need to know the fragment size.

If you don't want to put the value in this header, perhaps we could add another field to received data sample?

Reply:
The received data sample was one option I was considering.  Another would be to have the non-rtps transports set this value to 0 which would indicate to TransportReassembly that it should do the previous behavior

Reply:
I can make the two "split" cases in the reassembly logic dependent on a non-zero fragment size as well, but currently the only place where fragment ranges could overlap is from a non-OpenDDS implementation of RTPS. OpenDDS transports, in general, won't ever hit those two cases in the reassembly logic. So the check would be slightly redundant.

Reply:
Oh, this should have been removed. The fragment size is in ReceivedDataSample now.